### DISCUSSION TITLE
Commercial use

### DISCUSSION THREAD

**Discussion by ghost** [py-pdf/fpdf2#929] [Q&A] [answered]
I want ask about commercial use. This Python library can be used for commercials like hackathon or selling online?

**Comment by Lucas-C** [comment]
Hi @agusanjayaa

This library is under the GNU LGPL v3 license: https://github.com/py-pdf/fpdf2/blob/master/LICENSE

This website clearly states what can, and cannot be done with code under this license: https://www.tldrlegal.com/license/gnu-lesser-general-public-license-v3-lgpl-3

**Comment by ghost** [comment]
Hi @Lucas-C , thank you.

I have already read all the links but I still don't understand, I need a simple answer because I am afraid of violating the rules of this repository. That is why I asked in detail like commercial needs.

**Reply by gmischler** [reply]
> I have already read all the links

Are you sure?
The second link that @Lucas-C gave you tabulates in a very clear and concise way what can and cannot be done based on the given license.

(Besides that, the text of the actual license is the only authoratitative document. If you want legal certainty and don't think you understand the text of the license, then you need to aske a lawyer for advice. Any explanations by third parties (including us) are just opinions, and not legally binding.)

**Reply by Lucas-C** [reply, marked as answer]
> I have already read all the links but I still don't understand, I need a simple answer because I am afraid of violating the rules of this repository. That is why I asked in detail like commercial needs.

I agree with @gmischler.

Also, you did not really provide details on what you are planning to do exactly.

Organizing a hackathon where developpers would use `fpdf2` is totally fine of course 🙂

Overall the LGPL v3 license will let you use this library for commercial uses, but will require that you to disclose some of your code. Quoting the license terms:

> Convey the Minimal Corresponding Source under the terms of this License, and the Corresponding Application Code in a form suitable for, and under terms that permit, the user to recombine or relink the Application with a modified version of the Linked Version to produce a modified Combined Work
